_pages/extension_sharing_guidelines.md

```diff
@@ -0,0 +1,62 @@
+   1. LICENSE file detailing the licence for the NDX. Permissive licenses SHOULD be used if possible. BSD licence is RECOMMENDED. In particular for release via the NDX Catalog (see Sec. 4ii), the licence MUST include language giving permission to members of the GitHub organization to distribute the extensions.
+   1. Release notes detailing the changes between extension versions. Release notes MAY be included in the detailed RST documentation or as a separate RELEASENOTES.md/rst file.
+   1. Requirements file detailing dependencies on other NDX and where they can be installed from. This is required as the namespace.yaml only includes the names of namespaces that are being included.
+1. Releases of extensions SHOULD further include:
+   1. Source for generating the YAML specfication using the PyNWB/HDMF extensions APIs. By sharing the sources eases maintenance, testing, and reuse of the NDX.
+   1. Additional documentation in Sphinx RST format describing the NDX, credits, legal, and acknowledgements.
+   1. README file with additional details about the NDX.
+1. Releases of NDX MAY further include:
+   1. Custom Python plugins (e.g., Container classes for neurodata_types) for PyNWB. This is RECOMMENDED to make the NDX accessible to end users and has the advantage that custom functionality for interacting with the data can be defined.
+   1. OPTIONAL custom Matlab plugins for MatNWB for representing neurodata_types.
+1. Performing releases:
+   1. To ensure compliance with the semantic versioning rules, public NDX MUST be released in a persistent fashion (i.e., sources of a particular version MUST not be modified after release and MUST remain accessible).
+   1. Public NDX MUST register with the NWB:N extensions catalog. To register with the catalog, a public NDX MUST create a ndx-recipe Git repository as part of the NWB:N extension GitHub organization. An ndx-recipe repo contains the descriptions needed for the catalog and deployment, i.e. the sources of the extension MAY be hosted in a different public ndx-src repository.
+   1. Public ndx-src repositories SHOULD use Git tags to ensure compliance with (4i) and NDX SHOULD follow sharing strategies outlined in the NDX Catalogs guidelines and strategies for sharing extensions. Note, software versions managed via Git tags may differ from the version of the extensions namespace. However, both MUST follow semantic versioning rules.
+1. Naming namespaces and repositories:
+   1. Namespaces for extensions SHOULD generally follow the following naming conventions:
+      1. Use only lower-case ASCII letters (no special characters)
+      1. Use “-” to separate different parts of the name (no spaces allowed)
+      1. Use the naming schema “ndx-myname”, e.g., “ndx-cortical-surface”
```

Comment:
My preference for names would be:

* the namespace name is e.g. "speech" (`speech.namespace.yaml`)
* the repo name is e.g. "ndx-speech" (`git clone ...ndx-speech.git`)
* the package name is e.g. "ndx_speech" (`from ndx_speech import Transcription`)

We know all non-core namespaces are going to be neurodata extensions, so there isn't much point in having the ndx prefix there. It only serves a purpose for the repo and package names, where this will be alongside other repos and packages.

Reply:
I think we need to clarify this text a bit, because I think it is not clear when we refer to the name of the namespace.yaml file and name key of the namespace that is in the namespace.yaml file. My current thinking is:

* The 1) name key of the namespace, 2) the name of recipe repo in the NDX catalog, and 3) the name of package in python should all be ``ndx_speech``
* The name of 4) the yaml file of the namespace should be ``ndx_speech.namespace.yaml``

If I understand it correctly, you are proposing to change (4) the name of YAML namespace file to just ``speech.namespace.yaml`` and leave (1 -3) as is with ``ndx_speech``. I think that is probably fine. 

I'll update the text to indicate ``_`` instead of ``-`` as replacement for spaces to make sure we can have the same name for the python package and repos and namespace name.

Reply:
@bendichter I updated the text for "Naming namespaces and repositories". Does this look better?